### PR TITLE
LPS-198234, LPS-198232

### PR DESCRIPTION
**Pull request by gislayne-vitorino**
This is an update for: [LPS-198234](https://liferay.atlassian.net/browse/LPS-198234), [LPS-198232](https://liferay.atlassian.net/browse/LPS-198232)

### CODE REVIEW

**Comment by gislayne-vitorino** [comment]
ci:test:publications

**Comment by pei-jung** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/change/tracking/spi/reference/DDMFormInstanceTableReferenceDefinition.java [hunk 1]
```diff
@@ -71,6 +71,9 @@ public void defineChildTableReferences(
 						DDMFormInstance.class.getName())
 				)
 			)
+		).singleColumnReference(
+			DDMFormInstanceTable.INSTANCE.structureId,
+			DDMStructureTable.INSTANCE.structureId
```
Comment (pei-jung) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/change/tracking/spi/reference/DDMFormInstanceTableReferenceDefinition.java:76]: If you take a look at the `addFormInstance` methods in [DDMFormInstanceLocalServiceImpl](https://github.com/liferay/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMFormInstanceLocalServiceImpl.java), it never adds a DDMStructure when a new form instance is added. So DDMStructure is never a child of a form instance.

Also, the foreign key generally references a primary key in the parent table. In this case, `structureId` is the foreign key, so DDMStructure has to be the parent table.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/change/tracking/test/DDMFormInstanceTableReferenceDefinitionTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		return DDMFormInstanceTestUtil.addDDMFormInstance(
+			group, TestPropsValues.getUserId());
```
Comment (pei-jung) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/change/tracking/test/DDMFormInstanceTableReferenceDefinitionTest.java:38]: Adds the DDMStructure in the setUp and use the add that takes in a DDMStructure so it doesn't get added in the ctCollection.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/change/tracking/test/DDMFormInstanceRecordVersionTableReferenceDefinitionTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+				ServiceContextTestUtil.getServiceContext());
+	}
+
+	@Override
+	protected CTModel<?> addCTModel() throws Exception {
+		return _ddmFormInstanceRecordLocalService.updateFormInstanceRecord(
+			TestPropsValues.getUserId(),
+			_ddmFormInstanceRecord.getFormInstanceRecordId(), false,
+			_ddmFormInstanceRecord.getDDMFormValues(),
+			ServiceContextTestUtil.getServiceContext());
```
Comment (pei-jung) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/change/tracking/test/DDMFormInstanceRecordVersionTableReferenceDefinitionTest.java:64]: Use _ddmFormInstanceRecordLocalService.updateStatus so we don't add other unrelated entities in the ctCollection.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/change/tracking/spi/reference/DDMFormInstanceRecordVersionTableReferenceDefinition.java [hunk 4]
```diff
@@ -29,6 +30,11 @@ public class DDMFormInstanceRecordVersionTableReferenceDefinition
 	public void defineChildTableReferences(
 		ChildTableReferenceInfoBuilder<DDMFormInstanceRecordVersionTable>
 			childTableReferenceInfoBuilder) {
+
+		childTableReferenceInfoBuilder.assetEntryReference(
+			DDMFormInstanceRecordVersionTable.INSTANCE.
+				formInstanceRecordVersionId,
+			DDMFormInstanceRecord.class);
```
Comment (pei-jung) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/change/tracking/spi/reference/DDMFormInstanceRecordVersionTableReferenceDefinition.java:37]: Not needed. AssetEntry is not created when a DDMFormInstanceRecordVersion is added.

**Review by pei-jung — CHANGES_REQUESTED** [review]

**Comment by gislayne-vitorino** [comment]
ci:test:sf

**Review by pei-jung — APPROVED** [review]

**Comment by pei-jung** [comment]
ci:forward

**Comment by suilin** [comment]
The relevant doesn't contain the new integration tests, I will trigger the publications one to see the results.

**Comment by suilin** [comment]
ci:test:publications

**Comment by suilin** [comment]
ci:forward:force

**Pull request closed without merging**